lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingService.java

```diff
@@ -259,29 +262,53 @@ public JourneyResponse checkVcResponse(
         }
 
         if (configService.enabled(DL_AUTH_SOURCE_CHECK)
-                && getDcmawDlVc(newVcs).isPresent()
-                && getSuccessfulDlVc(sessionCredentials).isEmpty()) {
+                && requiresAuthoritativeSourceCheck(newVcs, sessionVcs)) {
             return JOURNEY_DL_AUTH_SOURCE_CHECK;
         }
 
         return JOURNEY_NEXT;
     }
 
-    private Optional<VerifiableCredential> getDcmawDlVc(List<VerifiableCredential> vcs) {
-        return vcs.stream()
-                .filter(vc -> Cri.DCMAW.equals(vc.getCri()))
-                .filter(
-                        vc ->
-                                vc.getCredential().getCredentialSubject()
-                                                instanceof PersonWithDocuments personWithDocuments
-                                        && personWithDocuments.getDrivingPermit() != null)
-                .findFirst();
+    private boolean requiresAuthoritativeSourceCheck(
+            List<VerifiableCredential> newVcs, List<VerifiableCredential> sessionVcs) {
+        return findSuccessfulVcFromCri(DCMAW, newVcs)
+                .map(this::getDrivingPermitIdentifier)
+                .map(
+                        dcmawDpId ->
+                                findSuccessfulVcFromCri(DRIVING_LICENCE, sessionVcs)
+                                        .map(
+                                                dlVc ->
+                                                        !Objects.equals(
+                                                                dcmawDpId,
+                                                                getDrivingPermitIdentifier(dlVc)))
+                                        .orElse(true))
+                .orElse(false);
     }
 
-    private Optional<VerifiableCredential> getSuccessfulDlVc(List<VerifiableCredential> vcs) {
+    private Optional<VerifiableCredential> findSuccessfulVcFromCri(
+            Cri cri, List<VerifiableCredential> vcs) {
         return vcs.stream()
-                .filter(vc -> Cri.DRIVING_LICENCE.equals(vc.getCri()))
+                .filter(vc -> cri.equals(vc.getCri()))
                 .filter(VcHelper::isSuccessfulVc)
                 .findFirst();
     }
+
+    private String getDrivingPermitIdentifier(VerifiableCredential vc) {
+        if (vc.getCredential().getCredentialSubject()
+                        instanceof IdentityCheckSubject identityCheckSubject
+                && identityCheckSubject.getDrivingPermit() != null
+                && !identityCheckSubject.getDrivingPermit().isEmpty()) {
+            var permit = identityCheckSubject.getDrivingPermit().get(0);
+            return String.format(
+                    "drivingPermit/%s/%s/%s/%s",
+                    permit.getIssuingCountry(),
+                    permit.getIssuedBy(),
+                    permit.getPersonalNumber(),
+                    permit.getIssueDate());
```

Comment:
Do you need to default these to 'unknown' (or anything else) if they are missing?

Reply:
They'll default to a liternal null if they're not present, which I think is fine for our purposes here